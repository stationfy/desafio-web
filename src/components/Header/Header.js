import React, { Component } from "react";
import { Link } from 'react-router-dom';

import './Header.css';

class Header extends Component {

  renderHeader(reponame) {
    if(!reponame) {
      return (
        <div className="header-title-block">
          <i className="fa fa-bars" aria-hidden="true" onClick={ () => { alert(`A menu here, soon...`) } }></i>
          <span className="header-title">Github JavaPop</span>
        </div>
      )
    } else {
      return (
        <div className="header-title-block">
          <Link to="/">
            <i className="fa fa-arrow-left header-back" aria-hidden="true"></i>
          </Link>
          <span className="header-title"> { reponame } </span>
        </div>
      )
    }
  }

  render() {
    const reponame = this.props.reponame;
    return (
      <div className="header content">
        { this.renderHeader(reponame) }
      </div>
    )
  }
}

export default Header