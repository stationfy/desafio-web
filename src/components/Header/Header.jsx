import React from "react";
import { withRouter } from "react-router-dom";

import PropTypes from "prop-types";

import ArrowIcon from "../../assets/arrow.png";
import { Container, Arrow, BurguerMenu, HeaderContainer } from "./Styles";

const Header = props => {
  const goHome = () => props.history.push("/");
  return (
    <Container>
      {props.isPull ? (
        <Arrow src={ArrowIcon} alt="Voltar" onClick={goHome} />
      ) : (
        <BurguerMenu>
          <div></div>
          <div></div>
          <div></div>
        </BurguerMenu>
      )}
      <HeaderContainer>
        <h1>{props.children}</h1>
      </HeaderContainer>
    </Container>
  );
};

Header.propTypes = {
  children: PropTypes.string.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func
  })
};

export default withRouter(Header);
