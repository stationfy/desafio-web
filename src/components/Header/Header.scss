.header {
  background-color: #343438;
  font-size: 1.2em;
  padding: 10px 50px;
  color: #ffffff;
  border-left: 2px solid #343438;
  border-right: 2px solid #343438;
}

.header-title-block {
  margin-left: 15px;
  padding: 35px 0px 5px;
}

.header-title {
  margin-left: 5px;
}

.header-back {
  color: #ffffff;
}
