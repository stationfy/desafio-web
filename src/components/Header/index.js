import React from "react";
import { withRouter } from 'react-router-dom';

import PropTypes from "prop-types";

import { Container } from "./Styles";

import { FaArrowLeft } from 'react-icons/fa';


const Header = props => {

  const homePage = () => props.history.push("/");

  const isHomePage = () => props.location.pathname === "/";

  return (
    <Container>
      {!isHomePage() ? <FaArrowLeft size={20} onClick={homePage} /> : null}
      <h1>{props.children}</h1>
    </Container>
  );
};

Header.propTypes = {
  children: PropTypes.string.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func
  })
};

export default withRouter(Header);
