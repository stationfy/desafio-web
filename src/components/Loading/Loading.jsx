import React from "react";

import LoadingIcon from "../../assets/loading.svg";

import { Container, Spinner } from "./Styles";

const Loading = () => {
  return (
    <Container>
      <Spinner src={LoadingIcon} alt="Carregando" />
    </Container>
  );
};

export default Loading;
