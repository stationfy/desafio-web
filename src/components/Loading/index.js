import React from "react";

import { Container, Spinner } from "./Styles";

const Loading = () => {
  return (
    <Container>
      <Spinner size={90} />
    </Container>
  );
};

export default Loading;
