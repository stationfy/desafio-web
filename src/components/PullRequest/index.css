.PullRequest-wrapper{
    width: 100%;
    height: 120px;
    border-bottom-style: solid;
    border-bottom-width: 0.5px;
    border-bottom-color: gray;
    margin-left: 10px;
    flex-direction: row;
    display: flex;
}

.PullRequest-infos{
    display: flex;
    flex: 1;
    flex-wrap: wrap;
    align-items: flex-start;
    flex-direction: column;
}

.PullRequest-user{
    flex: 4;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    flex-wrap: wrap;
    margin-bottom: 10px;
}

.user-img{
    width: 30px;
    border-radius: 50%;
    margin-bottom: 3px;
}

.username{
    color: blue;
    font-size: 15px;
    line-height: 0px;
    margin-left: 10px;
}

.name{
    color: gray;
    font-size: 13px;
    line-height: 0px;
    margin-top: -1px;
}

.PullRequest-title{
    font-size: 15px;
    color: blue;
}

.PullRequest-description{
    font-size: 13px;
    color: gray;
    width: 200px;
    height: 28px;
    white-space: wrap;
    overflow: hidden;
}

.PullRequest-user{
    display: flex;
    flex-direction: column;
    
}

.fork-number{
    color: #DE920C;
    font-size: 16px;
}

.star-number{
    color: #DE920C;
    font-size: 16px;
}

.PullRequest-status{
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: flex-start;
    margin-top: -12px;
}
