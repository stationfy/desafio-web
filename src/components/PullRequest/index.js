import React from "react";
import PropTypes from "prop-types";

import { Container } from "./Styles";

const handleDescription = (description) => !description ? "No description provided" : description.substring(0, 240) + '...';

const PullCard = ({
  pull: {
    title,
    body,
    user: { login, avatar_url }
  }
}) => {
  return (
    <Container>
      <h1>{title}</h1>
      <p>{handleDescription(body)}</p>
      <div>
        <img src={avatar_url} alt={`${login} avatar`} />
        <span>{login}</span>
      </div>
    </Container>
  );
};

PullCard.propTypes = {
  pull: PropTypes.shape({
    html_url: PropTypes.string,
    title: PropTypes.string,
    body: PropTypes.string,
    updated_at: PropTypes.string,
    user: PropTypes.shape({
      login: PropTypes.string,
      avatar_url: PropTypes.string
    })
  }).isRequired
};

export default PullCard;
