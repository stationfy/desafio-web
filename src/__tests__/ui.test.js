import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { mount, shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ReposPage from '../pages/ReposPage/ReposPage';
import PullsPage from '../pages/PullsPage/PullsPage';
import Header from '../components/Header';
import Loading from '../components/Loading';
import PullRequest from '../components/PullRequest';
import Repo from '../components/Repo';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });

describe('Components', () => {
  it('should render correctly <PullsPage>', () => {
    
    const mockStore = configureStore();
    const store = mockStore({
      pulls: {
        loading: false,
        data: [
          {
            id: 42,
            title: 'Tralala',
            body: 'Lorem ipsum avada mor',
            user: { login: 'Coralina', avatar_url: 'https://trelele.com.br' }
          }
        ]
      },
  });
    const wrapper = mount(
            <Provider store={store}>
              <MemoryRouter>
                <PullsPage match={{params:"/Coralina/TalkBot"}} />
              </MemoryRouter>
            </Provider>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly <ReposPage>', () => {
    
    const mockStore = configureStore();
    const store = mockStore({
      repos: {
        loading: false,
        data: [
          {
            id: 42,
            full_name: 'Captain Jack Sparrow',
            name: 'Sparrow',
            description: 'Owner of Black Pearl',
            owner: { login: 'CapJS', avatar_url: 'https://sparrowphotos.com' },
            stargazers_count: 3000,
            forks_count: 4000
          }
        ]
      },
  });
    const wrapper = mount(
            <Provider store={store}>
              <MemoryRouter>
                <ReposPage />
              </MemoryRouter>
            </Provider>);

    expect(wrapper).toMatchSnapshot();
  });
});

describe('Components', () => {
  it('should render correctly <Header />', () => {
    
    const wrapper = shallow(<Header />);
            
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly <Loading />', () => {
    
    const wrapper = shallow(<Loading />);
            
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly <PullRequest />', () => {
    
    const wrapper = shallow(
      <PullRequest 
      pull={
        {
         title: 'Tralala',
         body: 'Lorem ipsum avada mor',
         user: { login: 'Coralina', avatar_url: 'https://trelele.com.br' }
       }
     }
      />
    );
            
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly <Repo />', () => {
    
    const wrapper = shallow(
      <Repo 
      repo={
         {
          id: 42,
          full_name: 'Captain Jack Sparrow',
          name: 'Sparrow',
          description: 'Owner of Black Pearl',
          owner: { login: 'CapJS', avatar_url: 'https://sparrowphotos.com' },
          stargazers_count: 3000,
          forks_count: 4000
        }
      }
      />
    );
            
    expect(wrapper).toMatchSnapshot();
  });
});