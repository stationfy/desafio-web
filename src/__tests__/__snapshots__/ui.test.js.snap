// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components should render correctly <Header /> 1`] = `ShallowWrapper {}`;

exports[`Components should render correctly <Loading /> 1`] = `ShallowWrapper {}`;

exports[`Components should render correctly <PullRequest /> 1`] = `ShallowWrapper {}`;

exports[`Components should render correctly <PullsPage> 1`] = `ReactWrapper {}`;

exports[`Components should render correctly <Repo /> 1`] = `ShallowWrapper {}`;

exports[`Components should render correctly <ReposPage> 1`] = `ReactWrapper {}`;
